src/mcnntunes/minimizer.py

```diff
@@ -31,22 +32,65 @@ def __init__(self, runs, truth, model, output = None, truth_index = 0):
             self.write_on_disk = True
             self.output = output
 
+        # Compute degrees of freedom for chi2
+        self.dof = self.runs.weighted_dof
+
     def chi2(self, x):
         """Reduced chi2 estimator (weighted, eventually)"""
         x = x.reshape(1,self.runs.x_scaled.shape[1])
         prediction = self.model.predict(x, scaled_x = True, scaled_y = False)
-        return stats.chi2(prediction, self.truth, self.truth_error2, weights=self.runs.y_weight)
+        return stats.chi2(prediction, self.truth, self.truth_error2, weights=self.runs.y_weight, dof=self.dof)
 
     def unweighted_chi2(self, x):
         """Reduced chi2 estimator (always unweighted)"""
         x = x.reshape(1,self.runs.x_scaled.shape[1])
         prediction = self.model.predict(x, scaled_x = True, scaled_y = False)
-        return stats.chi2(prediction, self.truth, self.truth_error2)
+        return stats.chi2(prediction, self.truth, self.truth_error2, dof=self.runs.unweighted_dof)
 
     @abstractmethod
     def minimize(self):
         pass
 
+    def compute_errors(self, best_x_scaled):
+        """"""
+
+        N = 1000 # points
+        errors = []
+        best_x_unscaled = self.runs.unscale_x(best_x_scaled)
+        delta_chi2 = scipy.stats.chi2(self.dof).ppf(0.682689492137)  / self.dof
+
+        # Iterate over all params
+        for axis in range(self.runs.x_scaled.shape[1]):
+
+            # Scan the x-space and compute the chi2 profile
+            x_scan = np.linspace(np.min(self.runs.x_scaled[:, axis]),
+                                 np.max(self.runs.x_scaled[:, axis]),
+                                 N)
+            chi2_values = np.zeros(N)
+            x_unscaled  = np.zeros(N)
+            for index, param in enumerate(x_scan):
+                a = np.array(best_x_scaled)
+                a[axis] = param
+                chi2_values[index] = self.chi2(a)
+                x_unscaled[index]  = self.runs.unscale_x(a)[axis]
+
+            # Compute where chi2 - min(chi2) ~= delta_chi2
+            intersec_idxs = np.argwhere(np.diff(np.sign(
+                                chi2_values - np.min(chi2_values) - delta_chi2
+                            )))
```

Comment:
Possible caveat: if the ``min(chi2_values)`` is not equal to the minimum found by CMAES, we would obtain a weird result. However, this would underline a failure in the minimization process.